test/functional/normal/lang_spec.lua

```diff
@@ -0,0 +1,48 @@
+local helpers = require('test.functional.helpers')(after_each)
+local clear, insert, eq = helpers.clear, helpers.insert, helpers.eq
+local execute, expect = helpers.execute, helpers.expect
+local feed, eval = helpers.feed, helpers.eval
+
+describe('gu and gU', function()
+  before_each(clear)
+
+  it('works in any locale with default casemap', function()
+    eq('internal,keepascii', eval('&casemap'))
+    insert("iI")
+    feed("VgU")
+    expect("II")
+    feed("Vgu")
+    expect("ii")
+  end)
+
+  describe('works in Turkish locale', function()
+    before_each(function()
+      execute('lang ctype tr_TR.UTF-8')
```

Comment:
I would suggest to use `exc_exec` and checking for it returning zero, not for error message.